lib/collector_demon.rb

```diff
@@ -8,21 +8,23 @@ def self.prefix
   end
 
   def run
-    if @pid = fork
-      write_pid_file
-      return
-    end
+    @pid =
+      fork do
```

Comment:
```
static VALUE
rb_f_fork(VALUE obj)
{
    rb_pid_t pid;

    pid = rb_call_proc__fork();

    if (pid == 0) {
        if (rb_block_given_p()) {
            int status;
            rb_protect(rb_yield, Qundef, &status);
            ruby_stop(status);
        }
        return Qnil;
    }

    return PIDT2NUM(pid);
}
```

This is the ruby source code of `fork`. We see how the block is called with `rb_protect`. From what I've researched, `rb_protect` basically rescues all exceptions.